src/node_url.cc

```diff
@@ -829,6 +829,16 @@ static url_host_type ParseOpaqueHost(url_host* host,
   return type;
 }
 
+static inline bool IsAllASCII(std::string* input) {
```

Comment:
@mscdex  : I made changes to use existing contains_non_ascii.

Will check about folding checking and lowercasing into one.